mmv1/third_party/terraform/services/apigee/resource_apigee_api.go

```diff
@@ -0,0 +1,403 @@
+// ----------------------------------------------------------------------------
+//
+//     This file is partially automatically generated by Magic Modules and with manual
+//     changes to resourceApigeeApiCreate
+//
+// ----------------------------------------------------------------------------
+
+package apigee
+
+import (
+	"context"
+	"fmt"
+	"log"
+	"os"
+	"time"
+
+	"github.com/hashicorp/terraform-plugin-log/tflog"
+	"github.com/hashicorp/terraform-plugin-sdk/v2/helper/customdiff"
+	"github.com/hashicorp/terraform-plugin-sdk/v2/helper/schema"
+	"github.com/hashicorp/terraform-provider-google/google/tpgresource"
+	transport_tpg "github.com/hashicorp/terraform-provider-google/google/transport"
+)
+
+func ResourceApigeeApi() *schema.Resource {
+	return &schema.Resource{
+		Create: resourceApigeeApiCreate,
+		Read:   resourceApigeeApiRead,
+		Update: resourceApigeeApiUpdate,
+		Delete: resourceApigeeApiDelete,
+
+		Importer: &schema.ResourceImporter{
+			State: resourceApigeeApiImport,
+		},
+
+		CustomizeDiff: customdiff.All(
+			/*
+				If any of the config_bundle, detect_md5hash or md5hash is changed,
+				then an update is expected, so we tell Terraform core to expect update on meta_data,
+				latest_revision_id and revision
+			*/
+
+			customdiff.ComputedIf("meta_data", apigeeApiDetectBundleUpdate),
+			customdiff.ComputedIf("latest_revision_id", apigeeApiDetectBundleUpdate),
+			customdiff.ComputedIf("revision", apigeeApiDetectBundleUpdate),
+		),
+
+		Timeouts: &schema.ResourceTimeout{
+			Create: schema.DefaultTimeout(20 * time.Minute),
+			Update: schema.DefaultTimeout(20 * time.Minute),
+			Delete: schema.DefaultTimeout(20 * time.Minute),
+		},
+
+		Schema: map[string]*schema.Schema{
+			"name": {
+				Type:        schema.TypeString,
+				Required:    true,
+				ForceNew:    true,
+				Description: `Name of the API proxy. Restrict the characters used to: A-Za-z0-9._-.`,
```

Comment:
The description here should probably be:
The name of the API proxy. This field only accepts the following characters: ...